### PR TITLE
Comptime temp

### PR DESCRIPTION
**Pull request by GrayHatter**
More fixes for the comptime directive implementation

### CODE REVIEW

File: src/template/page.zig [hunk 1]
```diff
@@ -339,173 +409,141 @@ pub fn Page(comptime template: Template, comptime PageDataType: type) type {
...
-                    },
-                    else => unreachable,
-                }
-                if (vari) |v| {
-                    try out.writeAll(v);
-                }
-            } else {
-                try offset.kind.directive.formatTyped(T, data, out);
-            }
+        fn vecCount(ofs: []const Offset) usize {
```
Comment (e0ff) [inline src/template/page.zig]: This function isn't used.

File: src/template/page.zig [hunk 2]
```diff
@@ -126,26 +141,43 @@ test getOffset {
...
-                else => unreachable,
+            switch (f.type) {
+                []const u8 => unreachable,
+                ?[]const u8 => unreachable,
+                ?usize => unreachable,
+                else => switch (@typeInfo(f.type)) {
+                    .Pointer => |ptr| return ptr.child,
+                    .Optional => |opt| return opt.child,
+                    .Struct => return f.type,
+                    .Int => return f.type,
```
Comment (e0ff) [inline src/template/page.zig:162]: What about float, enum, and union?

File: src/template/page.zig [hunk 3]
```diff
@@ -339,173 +409,141 @@ pub fn Page(comptime template: Template, comptime PageDataType: type) type {
...
-                            }
-                        },
-                    };
-                }
-            } else {
-                std.debug.print("error generating page, field {s} is missing\n", .{name});
+        fn offsetOptionalItem(T: type, item: ?T, comptime ofs: []const Offset, html: []const u8, out: anytype) !void {
+            if (comptime T == ?[]const u8) return offsetDirective(T, item.?, ofs[0], out);
+            switch (@typeInfo(T)) {
+                .Int => std.debug.print("skipped int\n", .{}),
```
Comment (e0ff) [inline src/template/page.zig:501]: What about float, enum, and union?

File: src/template/page.zig (hunk 2 above)
Comment (GrayHatter) [reply src/template/page.zig:162]: those are currently impossible type for the template compiler to generate

File: src/template/page.zig (hunk 3 above)
Comment (GrayHatter) [reply src/template/page.zig:501]: both should be impossible, I'll make the following `else => unreachable` into `else => comptime unreachable` to prove that there's no way for any to reach this point in the code.

File: src/template/page.zig (hunk 1 above)
Comment (GrayHatter) [reply src/template/page.zig]: fixed in iovec pr

**Review by e0ff — APPROVED** [review]

**Pull request merged**